public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchRequest.ts

```diff
@@ -93,23 +95,35 @@ export abstract class CloudWatchRequest {
     }, {});
   }
 
+  isMultiVariable(target?: string) {
+    if (target) {
+      const variables = this.templateSrv.getVariables();
+      const variable = variables.find(({ name }) => name === getVariableName(target));
+      const type = variable?.type;
+      return (type === 'custom' || type === 'query' || type === 'datasource') && variable?.multi;
+    }
+
+    return false;
+  }
+
+  isVariableWithMultipleOptionsSelected(target?: string, scopedVars?: ScopedVars) {
+    if (!target || !this.isMultiVariable(target)) {
+      return false;
+    }
+    return this.expandVariableToArray(target, scopedVars || {}).length > 1;
+  }
+
   replaceVariableAndDisplayWarningIfMulti(
     target?: string,
     scopedVars?: ScopedVars,
     displayErrorIfIsMultiTemplateVariable?: boolean,
     fieldName?: string
   ) {
-    if (displayErrorIfIsMultiTemplateVariable && !!target) {
-      const variables = this.templateSrv.getVariables();
-      const variable = variables.find(({ name }) => name === getVariableName(target));
-      const isMultiVariable =
-        variable?.type === 'custom' || variable?.type === 'query' || variable?.type === 'datasource';
-      if (isMultiVariable && variable.multi) {
-        this.debouncedCustomAlert(
-          'CloudWatch templating error',
-          `Multi template variables are not supported for ${fieldName || target}`
-        );
-      }
+    if (displayErrorIfIsMultiTemplateVariable && this.isVariableWithMultipleOptionsSelected(target)) {
```

Comment:
This now checks if a template variable is both a multi-enabled variable and if multiple options have been selected for it.